tests/unittests/algo/test_tpe.py

```diff
@@ -262,13 +370,13 @@ def test_set_state(self, tpe):
     def test_unsupported_space(self):
         """Test tpe only work for supported search space"""
         space = Space()
-        dim = Integer('yolo1', 'uniform', -2, 4)
+        dim = Fidelity('epoch', 1, 9, 3)
```

Comment:
It could support easily Fidelity by just returning the max like random search does. `Integer('yolo1', 'normal', 0, 1)` for testing unsupported priors.

Reply:
It is pretty straight forward for random as all suggestions simply follow the same pattern, for tpe, we need to add separate paths to handle fidelity dimension and mention such support in doc. The only value for random or tpe to support fidelity I can think is to make it easier for user to switch from different search algorithms, or user can simply specify fixed fidelity value for random and tpe.

Reply:
Yes the advantage of supporting fidelity dimension in algos like random search and TPE is really just to make it easier to switch from one algo to another using the same search space definition.